src/compiler/gosling-to-higlass.ts

```diff
@@ -152,10 +154,19 @@ export function goslingToHiGlass(
                 x1: getFieldName('x1'),
                 x1e: getFieldName('x1e')
             } as const;
-            // use gosling's custom data fetchers
+
+            // Check whether there are any URL specific fetch options
+            const urlFetchOptions =
+                ('url' in firstResolvedSpec.data && urlToFetchOptions?.[firstResolvedSpec.data.url]) || {};
+            const indexUrlFetchOptions =
+                ('indexUrl' in firstResolvedSpec.data && urlToFetchOptions?.[firstResolvedSpec.data.indexUrl]) || {};
+
+            // This object will be passed to the data fetchers
             hgTrack.data = {
                 ...firstResolvedSpec.data,
                 ...xFields,
+                urlFetchOptions,
+                indexUrlFetchOptions,
```

Comment:
This is where the `urlToFetchOptions` map is used to pass the fetch options to the data fetchers